pkg/justification/processor_test.go

```diff
@@ -179,6 +190,88 @@ func TestCreateToken(t *testing.T) {
 			},
 			wantErr: "must be less than 1000 bytes",
 		},
+		{
+			name: "happy_path_with_validator",
+			request: &jvspb.CreateJustificationRequest{
+				Justifications: []*jvspb.Justification{
+					{
+						Category: "explanation",
+						Value:    "test",
+					},
+				},
+				Ttl: durationpb.New(3600 * time.Second),
+			},
+			validators: map[string]jvspb.Validator{
+				"explanation": &mockValidator{
+					resp: &jvspb.ValidateJustificationResponse{
+						Valid: true,
+					},
+				},
+			},
+			wantTTL:       1 * time.Hour,
+			wantAudiences: []string{DefaultAudience},
+		},
+		{
+			name: "happy_path_with_irrelevant_validator",
+			request: &jvspb.CreateJustificationRequest{
+				Justifications: []*jvspb.Justification{
+					{
+						Category: "explanation",
+						Value:    "test",
+					},
+				},
+				Ttl: durationpb.New(3600 * time.Second),
+			},
+			validators: map[string]jvspb.Validator{
+				"jira": &mockValidator{
+					resp: &jvspb.ValidateJustificationResponse{
+						Valid: false,
+						Error: []string{"bad jira ticket"},
+					},
+				},
+			},
+			wantTTL:       1 * time.Hour,
+			wantAudiences: []string{DefaultAudience},
+		},
+		{
+			name: "failed_validator_criteria",
+			request: &jvspb.CreateJustificationRequest{
+				Justifications: []*jvspb.Justification{
+					{
+						Category: "explanation",
+						Value:    "test",
+					},
+				},
+				Ttl: durationpb.New(3600 * time.Second),
+			},
+			validators: map[string]jvspb.Validator{
+				"explanation": &mockValidator{
+					resp: &jvspb.ValidateJustificationResponse{
+						Valid: false,
+						Error: []string{"bad explanation"},
+					},
+				},
+			},
+			wantErr: "failed validation criteria with error [bad explanation] and warning []",
+		},
+		{
+			name: "validator_err",
+			request: &jvspb.CreateJustificationRequest{
+				Justifications: []*jvspb.Justification{
+					{
+						Category: "explanation",
+						Value:    "test",
+					},
+				},
+				Ttl: durationpb.New(3600 * time.Second),
+			},
+			validators: map[string]jvspb.Validator{
+				"explanation": &mockValidator{
+					err: fmt.Errorf("Cannot connect to validator"),
+				},
+			},
+			wantErr: "unexpected error from validator \"explanation\": Cannot connect to validator",
+		},
```

Comment:
Can you add a test where no validator exists for the category?

Reply:
Added test case `missing_validator`